### PR TITLE
Implement Vue integration for Abby feature flagging and experimentation

### PR DESCRIPTION
**Pull request by feliciien**
Title: Implement Vue integration for Abby feature flagging and experimentation

Description: This pull request adds the Vue integration for Abby, ensuring feature parity with existing integrations and consuming @tryabby/core internally for feature flagging and experimentation. The integration includes useAbby, useFeatureFlag, and useRemoteConfig functionalities, along with tests for code quality and functionality verification.

Changes:

Added Vue integration in the @tryabby/vue directory
Implemented useAbby, useFeatureFlag, and useRemoteConfig functions
Ensured consumption of @tryabby/core for feature flagging and experimentation
Added tests for the Vue integration functionalities
This contribution aims to enhance Abby's compatibility with Vue projects and improve the overall feature flagging and experimentation capabilities.

/attempt #68

Thank you for the opportunity to work on this bounty!

/claim #68
@coderabbitai review

## Summary by CodeRabbit

- **New Features**
  - Introduced Vue integration for feature flagging and experimentation with Abby.
  - Added Vue composition functions for A/B testing, managing feature flags, and fetching remote configuration values.
  - Configurations added for launching and debugging applications in Node.js, Go, and Chrome.

- **Tests**
  - Added unit tests for new Vue composition functions ensuring correct functionality and integration.

- **Documentation**
  - Updated project settings to ignore non-essential files and directories in the development environment.

- **Refactor**
  - Set up TypeScript compilation options tailored for Vue projects.

### CODE REVIEW

File: .vscode/easycode.ignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+cache/
+.*/
+*.min.*
+*.test.*
+*.spec.*
+*.bundle.*
+*.bundle-min.*
+*.*.js
+*.*.ts
+*.log
```
Comment (cstrnt) [inline .vscode/easycode.ignore:13]: Do we need this file here? What is easycode?

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            ],
+            "runtimeExecutable": "npm",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "type": "node"
+        },
+        {
+            "name": "Attach to Process",
+            "type": "go",
```
[the remark this answers is not in the document]
Comment (cstrnt) [reply .vscode/launch.json:22]: I think this comment is valid

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        },
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Lancer Chrome en utilisant localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (cstrnt) [inline .vscode/launch.json:35]: The comments and names should be in English

File: packages/packages/vue/@tryabby/vue/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@tryabby/vue",
+  "version": "1.0.0",
+  "description": "Vue integration for Abby feature flagging and experimentation",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest"
+  },
```
Comment (cstrnt) [inline packages/packages/vue/@tryabby/vue/package.json:8]: We need a build script here in order to package and bundle this

File: packages/packages/vue/@tryabby/vue/src/useAbby.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// packages/vue/@tryabby/vue/src/useAbby.ts
```
Comment (cstrnt) [inline packages/packages/vue/@tryabby/vue/src/useAbby.ts:1]: Is there any particular reason why we are having those comments at the top of those files?

File: packages/packages/vue/@tryabby/vue/src/useAbby.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+import { ref, onMounted } from 'vue';
+import { getVariant, onAct } from '@tryabby/core';
```
Comment (cstrnt) [inline packages/packages/vue/@tryabby/vue/src/useAbby.ts:3]: this is **not** how the Abby core works. The core exports a class that needs to be initialized and then you can call those functions. Please have a look at the react implementation to see how this is done

**Review by cstrnt — CHANGES_REQUESTED** [review]
First of all thank you very much for your work!

Besides my comments here are some things that need to be addressed:

- Folder Structure: the package should live inside of `packages/vue`
- Testing: Abby uses vitest instead of jest for unit tests
- Reactivity: The current implementation isn't reactive at all, meaning that updates to flags wouldn't be registered
- Functionality: This just doesn't do anything yet. No data is loaded from the Abby API thus resulting in not doing anything
- Bundling & Exports: We need to export the created hooks in order to publish it to npm. Please have a look at the existing packages as a reference. We should export a function like `createAbby` which takes in the config
- TypeSafety/DX: The current implementation doesn't use any generics that fulfil the TypeSafety standard of Abby.

Please let me know if you need some help with this :) I know this is a lot and quite a lot of business knowledge, but I can just recommend using the React integration as a template/starting point

**Comment by cstrnt** [comment]
@feliciien could you please reply here if you are gonna change something or continue working on this? otherwise I would close this PR

**Comment by feliciien** [comment]
Yes I will continue working on it

**Pull request closed without merging**